clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignor.java

```diff
@@ -178,6 +178,21 @@ public GroupAssignment(Map<String, Assignment> assignments) {
         }
     }
 
+    /**
+     * {@link RebalanceProtocol#EAGER} rebalance protocol will let a consumer to always revoke all its owned partitions
+     * before participating the rebalance event, and therefore allows a complete reshuffling of the assignment.
+     *
+     * {@link RebalanceProtocol#COOPERATIVE} rebalance protocol will let a consumer to retain its currently assigned
+     * partitions before participating the rebalance event, as a result the assignor would not reassign its owned
+     * partitions immediately but may only indicate it to revoke first before the next rebalance event, and therefore is
+     * suitable for cooperative adjustments of assignments.
+     * {@link ConsumerPartitionAssignor} implementors who claim to support {@link RebalanceProtocol#COOPERATIVE} protocol
+     * from {@link ConsumerPartitionAssignor#supportedProtocols()} needs to make sure its
+     * {@link ConsumerPartitionAssignor#assign(Cluster, GroupSubscription)} logic respects the specified
+     * {@link Subscription#ownedPartitions()} list and not reassign them to other consumers; instead it can decide
```

Comment:
"... and not reassign them to other consumers..." -> "... and does not reassign them to other consumers until they have been revoked. In other words, any partitions that are being revoked from their current owner during this rebalance should be removed from the new assignment as well, and wait for the next rebalance to be reassigned." WDYT?

the phrasing "it can decide" makes it seem optional, not sure if intentional or not?

Reply:
Sounds good!